src/main/java/bio/terra/service/auth/iam/IamProviderInterface.java

```diff
@@ -130,6 +130,18 @@ Map<IamRole, String> createSnapshotResource(
       AuthenticatedUserRequest userReq, UUID snapshotId, SnapshotRequestModelPolicies policies)
       throws InterruptedException;
 
+  /**
+   * Create a snapshot builder request IAM resource
+   *
+   * @param userReq authenticated user
+   * @param snapshotId id of the snapshot the request is based off of
+   * @param snapshotBuilderRequestId id of the snapshot builder request
+   * @return Map of policy group emails for the snapshot builder request policies
+   */
+  Map<IamRole, List<String>> createSnapshotBuilderRequestResource(
```

Comment:
I'm a bit confuse on the difference between creating a `snapshot builder request` and `snapshot request`. Is `snapshotBuilderRequest` a request to a data custodian to build a snapshot, and `snapshot request` is after the building is done?

Reply:
`snapshot builder request` is a request from the snapshot builder to a custodian to build a snapshot. `SnapshotRequest` is probably technically better, but I want to leave this method with its current name, cause that is what the ResourceType in Sam is called.